kubeflow/jupyter/tests/jupyter_test.py

```diff
@@ -112,19 +112,17 @@ def test_jupyter(env, namespace):
   host = api_client.configuration.host
   logging.info("Kubernetes master: %s", host)
   master = host.rsplit("/", 1)[-1]
-
-  this_dir = os.path.dirname(__file__)
-  app_dir = os.path.join(this_dir, "test_app")
-
-  ks_cmd = ks_util.get_ksonnet_cmd(app_dir)
-
-  name = "jupyter-test"
   service = "jupyter-test"
-  component = "jupyter"
-  params = ""
-  ks_util.setup_ks_app(app_dir, env, namespace, component, params)
 
-  util.run([ks_cmd, "apply", env, "-c", component], cwd=app_dir)
+  with open("jupyter_test.yaml") as params: 
+    wf_result = yaml.load(params)
+    namespace = wf_result['metadata']['namespace']
+    name = wf_result['metadata']['name']
+    k8s_apps_v1 = k8s_client.AppsV1Api()
+    resp = k8s_apps_v1.create_namespaced_deployment(
```

Comment:
This code doesn't look right. Why are you creating a deployment here? The  contents of jupyter_test.yaml is a Notebook so you need to use the custom resource APi.

You can take a look at how we handle Argo workflows
https://github.com/kubeflow/testing/blob/f1d5dcd63b91f9b327beaf1e3aa48543f7c00ad5/py/kubeflow/testing/run_e2e_workflow.py#L357
We need to do something similar.